src/backend/distributed/commands/utility_hook.c

```diff
@@ -616,11 +666,88 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 		if (isAlterExtensionUpdateCitusStmt)
 		{
 			citusCanBeUpdatedToAvailableVersion = !InstalledAndAvailableVersionsSame();
+
+			/*upgrade citus: alter extension citus update to 'xxx' */
+			DefElem *newVersionValue = GetExtensionOption(
+				((AlterExtensionStmt *) parsetree)->options, "new_version");
+			Oid citusColumnarOid = get_extension_oid("citus_columnar", true);
+			if (newVersionValue)
+			{
+				char *newVersion = defGetString(newVersionValue);
+				double newVersionNumber = GetExtensionVersionNumber(strdup(newVersion));
+
+				/*alter extension citus update to version >= 11.1-1, and no citus_columnar installed */
+				if (newVersionNumber * 100 >= 1110 && citusColumnarOid == InvalidOid)
```

Comment:
since we consistently do * 100, should we change GetExtensionVersionNumber to return an int?

Reply:
done